neqo-transport/src/send_stream.rs

```diff
@@ -1369,6 +1380,31 @@ impl SendStream {
     pub(crate) fn state(&mut self) -> &mut SendStreamState {
         &mut self.state
     }
+
+    pub(crate) fn maybe_emit_writable_event(
+        &mut self,
+        previous_limit: usize,
+        current_limit: usize,
+    ) {
+        let low_watermark = self.writable_event_low_watermark.into();
+
+        if low_watermark < previous_limit {
+            // Stream was not constrained by limit before.
+            return;
+        }
+
+        if current_limit < low_watermark {
+            // Stream is still constrained by limit.
+            return;
+        }
+
+        if self.avail() < low_watermark {
+            // Stream is constrained by different limit.
+            return;
+        }
+
+        self.conn_events.send_stream_writable(self.stream_id);
```

Comment:
I find the current version, using early-return, easier to read, especially since it separates the 3 cases (_not constrained by limit_, _still constrained by limit_, _constrained by different limit_).

@martinthomson do you feel strongly about squashing it all together into 3 lines?

Reply:
No, but I found the verbosity unnecessary.